### PR TITLE
change: Regenerate Tests

### PR DESCRIPTION
**Pull request by pgaspar**
Update `change` tests to: `1.3.0 258c807`

Update generator to use the standard error indicator.
Update solution to pass updated tests.

### CODE REVIEW

File: exercises/change/change_test.rb [hunk 1]
```diff
@@ -45,16 +45,22 @@ def test_no_coins_make_0_change
 
   def test_error_testing_for_change_smaller_than_the_smallest_of_coins
     skip
-    assert_equal -1, Change.generate([5, 10], 3)
+    assert_raises(ArgumentError) do
+      Change.generate([5, 10], 3)
```
Comment (Insti) [inline exercises/change/change_test.rb]: `ArgumentError` is probably OK here, but what about returning a subclass which is more descriptive?

File: exercises/change/change_test.rb [hunk 2]
```diff
@@ -45,16 +45,22 @@ def test_no_coins_make_0_change
...
+    end
   end
 
   def test_error_if_no_combination_can_add_up_to_target
     skip
-    assert_equal -1, Change.generate([5, 10], 94)
+    assert_raises(ArgumentError) do
+      Change.generate([5, 10], 94)
```
Comment (Insti) [inline exercises/change/change_test.rb]: `ArgumentError` is probably not appropriate here.
Maybe `nil` or a custom error?

File: exercises/change/change_test.rb [hunk 3]
```diff
@@ -45,16 +45,22 @@ def test_no_coins_make_0_change
...
+    end
   end
 
   def test_cannot_find_negative_change_values
     skip
-    assert_equal -1, Change.generate([1, 2, 5], -5)
+    assert_raises(ArgumentError) do
+      Change.generate([1, 2, 5], -5)
```
Comment (Insti) [inline exercises/change/change_test.rb]: `ArgumentError` is reasonable here.

(I don't think invalid input like this is something we should be testing for, but that's an unrelated issue. (and very much a matter of opinion.))

**Comment by pgaspar** [comment]
Hello @iHiD! Thank you for your review. Wanted us to deal with #870 before this one, that's why I took so long to answer.

I understand and agree with your points, but I have a question:

Given how all 3 of these "error cases" are specified on the canonical data as `"expected": {"error": ...}`, how do you typically differentiate between them in the `change_case.rb` file? Should I check for the test name using something [like this](https://github.com/exercism/ruby/blob/master/exercises/alphametics/.meta/generator/alphametics_case.rb#L9)?

**Comment by Insti** [comment]
I assume you meant @Insti there :)

Yeah you'll need to do something like that. Either check the test name or error text I guess.

**Comment by iHiD** [comment]
Ah! I was confused :)

**Comment by pgaspar** [comment]
Oh no! Sorry @iHiD 😅 Wrote that in a hurry and didn't notice it!

@Insti Thanks :+1: I'll work on something soon.

**Comment by pgaspar** [comment]
@Insti I just pushed some tentative changes. I didn't follow your suggestions to the letter, but wanted to get your opinion regardless.

Please let me know what you think. I also appreciate suggestions for better error names :+1:

File: exercises/change/.meta/solutions/change.rb [hunk 4]
```diff
@@ -1,18 +1,23 @@
 class Change
   attr_reader :coins, :target
 
+  class NegativeTargetError < ArgumentError; end
+  class ImpossibleCombinationError < StandardError; end
```
Comment (Insti) [inline exercises/change/.meta/solutions/change.rb]: I really like these two :+1:

**Comment by Insti** [comment]
Do you think this is ready to merge now @pgaspar ?

**Comment by pgaspar** [comment]
I think so @Insti! Thank you 👍

**Comment by Insti** [comment]
Thanks for all your work on these @pgaspar :heart:

**Pull request merged**